integration/securitypolicy_test.go

```diff
@@ -0,0 +1,178 @@
+				ad.Name = api1Name
+				ad.Spec.Name = api1Name
+				ad.Spec.Proxy.ListenPath = "/test-api-1"
+			})
+			is.NoErr(err)
+
+			_, err = createTestAPIDef(ctx, c, testNs, func(ad *v1alpha1.ApiDefinition) {
+				ad.Name = api2Name
+				ad.Spec.Name = api2Name
+				ad.Spec.Proxy.ListenPath = "/test-api-2"
+			})
+			is.NoErr(err)
+
+			// ensure API is created on Tyk before creating policy
+			apiDef := &v1alpha1.ApiDefinition{ObjectMeta: metav1.ObjectMeta{Name: api1Name, Namespace: testNs}}
+			err = waitForTykResourceCreation(c, apiDef)
+			is.NoErr(err)
+
+			_, err = createTestPolicy(ctx, testNs, func(policy *v1alpha1.SecurityPolicy) {
+				policy.Name = policyName
+				policy.Spec.Name = policyName + testNs
+				policy.Spec.AccessRightsArray = []*v1alpha1.AccessDefinition{{Name: api1Name, Namespace: testNs}}
+			}, c)
+			is.NoErr(err)
+
+			pol := v1alpha1.SecurityPolicy{ObjectMeta: metav1.ObjectMeta{Name: policyName, Namespace: testNs}}
+			err = waitForTykResourceCreation(c, &pol)
+			is.NoErr(err)
+
+			return ctx
+		}).Assess("validate links are created properly",
+		func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			var pol v1alpha1.SecurityPolicy
+			var api v1alpha1.ApiDefinition
+
+			testNs, ok := ctx.Value(ctxNSKey).(string)
+			is.True(ok)
+
+			// check status of policy
+			err := c.Client().Resources().Get(ctx, policyName, testNs, &pol)
+			is.NoErr(err)
+
+			is.True(len(pol.Status.LinkedAPIs) != 0)
+			is.Equal(pol.Status.LinkedAPIs[0].Name, api1Name)
+
+			// check status of ApiDefinition
+			err = c.Client().Resources().Get(ctx, api1Name, testNs, &api)
+			is.NoErr(err)
+
+			is.True(len(api.Status.LinkedByPolicies) != 0)
+			is.Equal(api.Status.LinkedByPolicies[0].Name, policyName)
+
+			return ctx
+		}).Assess("Add new api in the access rights",
+		func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			var updatePolicy v1alpha1.SecurityPolicy
+			testNs, ok := ctx.Value(ctxNSKey).(string)
+			is.True(ok)
+
+			err := c.Client().Resources().Get(ctx, policyName, testNs, &updatePolicy)
+			is.NoErr(err)
+
+			updatePolicy.Spec.AccessRightsArray = append(updatePolicy.Spec.AccessRightsArray,
+				&v1alpha1.AccessDefinition{Name: api2Name, Namespace: testNs})
+
+			err = c.Client().Resources().Update(ctx, &updatePolicy)
+			is.NoErr(err)
+
+			var pol v1alpha1.SecurityPolicy
+			pol.Name = policyName
+			pol.Namespace = testNs
+
+			// wait until status of policy is updated.
+			err = wait.For(conditions.New(c.Client().Resources()).ResourceMatch(&pol, func(object k8s.Object) bool {
+				pol, ok := object.(*v1alpha1.SecurityPolicy)
+				if !ok {
+					return false
+				}
+
+				if len(pol.Status.LinkedAPIs) == 2 {
+					return true
+				}
+
+				return false
+			}), wait.WithTimeout(defaultWaitTimeout), wait.WithInterval(defaultWaitInterval))
+			is.NoErr(err)
+
+			is.True(pol.Status.LinkedAPIs[0].Name == api2Name || pol.Status.LinkedAPIs[1].Name == api2Name)
```

Comment:
here, we only check whether CR is updated or not. should we check that updating `access_rights_array` affects the Security Policy on Tyk? What do you think?

Reply:
I didn't add that because I thought we should have a separate tests to check policies are getting updated on Tyk.
I think you added that in your tests if I remember correctly.